### PR TITLE
Documentation changes based on dashboard updates

### PR DESCRIPTION
**Pull request by Greggless**
Here are the proposed changes to docs.snapccraft to facilitate the changes happening in dashboard

The architecture will follow this pattern:
- /build-snaps/publish
  - /build-snaps/publish/register-name
  - /build-snaps/publish/upload
  - /build-snaps/publish/release

I will be creating pages for the last three items on that list

### CODE REVIEW

File: build-snaps/publish.md [hunk 1]
```diff
@@ -3,11 +3,18 @@ layout: base
 title: Publish your snap
 ---
 
-You can share your snaps with the world by publishing them to the Snap Store. Alternatively, you can publish to a [brand store](/core/store).
+You can share your snaps with the world by publishing them to the Snap Store. Alternatively, you can publish to a [brand store](/core/store). 
 
-## 1. Create a Snap Store account
+There are three key stages to publishing a snap:
+1. [Registering a name for you snap](https://docs.snapcraft.io/build-snaps/register-name)
```
Comment (caldav) [inline build-snaps/publish.md]: These should be relative links

File: build-snaps/publish.md [hunk 2]
```diff
@@ -3,11 +3,18 @@ layout: base
...
+2. [Uploading your snap](https://docs.snapcraft.io/build-snaps/upload)
```
Comment (caldav) [inline build-snaps/publish.md]: Absolute -> relative link

File: build-snaps/publish.md [hunk 3]
```diff
@@ -3,11 +3,18 @@ layout: base
...
+3. [Releasing your snap](https://docs.snapcraft.io/build-snaps/release)
```
Comment (caldav) [inline build-snaps/publish.md]: Absolute -> relative link

File: build-snaps/publish.md [hunk 4]
```diff
@@ -23,97 +30,5 @@ Make sure the `snapcraft` and `snap` commands know about you by logging in using
...
-    beta       -          -
-    edge       -          -
-
-Your snap is now available for users to install and in case of a `stable` release, it will be listed in search results.
-
-    $ snap find drone
-    Name             Version   Developer  Notes  Summary
-    drone-autopilot  stableV2  you        -      An autopilot mode for your drones
+## What comes next?
+[Registering a name for your snap](https://docs.snapcraft.io/build-snaps/register-name)
```
Comment (caldav) [inline build-snaps/publish.md]: Absolute -> relative link

**Comment by Greggless** [comment]
All updated! Thanks

**Pull request merged**